src/js/actions/index.js

```diff
@@ -102,13 +102,33 @@ export function hideUserActions() {
   };
 }
 
-export function fetchReasonCodes() {
-  const url = '/api/reasonCodes';
+export function fetchReasonCodes(activityCode = null, dispatchType = FETCH_REASONCODES) {
+  let url = '/api/reasonCodes';
+  if (activityCode) {
+    url += `?activityCode=${activityCode}`;
+  }
+
   return (dispatch) => {
     apiClient.get(url).then((res) => {
+      if (res.data) {
+        // These reason codes are exclusively used in dropdown selectors, so for the sake of
+        // convenience, we immediately transform the response to conform to that format.
+        const reasonCodes = res.data.data.map((reasonCode) => (
+          {
+            id: reasonCode.id,
+            value: reasonCode.id,
+            label: reasonCode.name,
+          }
+        ));
+        dispatch({
+          type: dispatchType,
+          payload: reasonCodes,
+        });
+        return;
+      }
       dispatch({
-        type: FETCH_REASONCODES,
-        payload: res.data,
+        type: dispatchType,
+        payload: [],
```

Comment:
I think you should be able to do something like this:
```
        const reasonCodes = res?.data?.data?.map?.((reasonCode) => (
          {
            id: reasonCode.id,
            value: reasonCode.id,
            label: reasonCode.name,
          }
        ));
        dispatch({
          type: dispatchType,
          payload: reasonCodes || [],
        });
```
When something from the call chain is null/undefined, it will fall back to [].

Note:
`||` use the fallback when the checked value is falsy,
`??` use the fallback when the checked value is null.
In this case, I think `reasonCodes` will fall back to `null`.